File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cur_big = 0;
        int big = INT_MIN;
        for (int i=0; i<nums.size(); i++){
            cur_big = max(nums[i], nums[i] + cur_big);
            big = max(big, cur_big);
        }
        return big;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cur_max = nums[0]
        global_max = nums[0]
        for i in range(1,len(nums)):
            cur_max = max(nums[i], cur_max+nums[i])
            global_max = max(global_max, cur_max)
        return global_max
        